src/core/layout/qgslayoutitemmapgrid.cpp

```diff
@@ -1894,6 +1899,22 @@ bool QgsLayoutItemMapGrid::shouldShowForDisplayMode( QgsLayoutItemMapGrid::Annot
          || ( mode == QgsLayoutItemMapGrid::LongitudeOnly && coordinate == QgsLayoutItemMapGrid::Longitude );
 }
 
+
+QgsLayoutItemMapGrid::DisplayMode gridAnnotationDisplayModeFromDD( QString ddValue, QgsLayoutItemMapGrid::DisplayMode defValue )
+{
+  if ( ddValue.toLower() == QStringLiteral( "x_only" ) )
+    return QgsLayoutItemMapGrid::LatitudeOnly;
+  else if ( ddValue.toLower() == QStringLiteral( "y_only" ) )
+    return QgsLayoutItemMapGrid::LongitudeOnly;
+  else if ( ddValue.toLower() == QStringLiteral( "disabled" ) )
+    return QgsLayoutItemMapGrid::HideAll;
```

Comment:
```suggestion
  if ( ddValue.compare( QLatin1String( "x_only" ), Qt::CaseInsensitive ) == 0 )
    return QgsLayoutItemMapGrid::LatitudeOnly;
  else if ( ddValue.compare( QLatin1String( "y_only", Qt::CaseInsensitive ) == 0 )
    return QgsLayoutItemMapGrid::LongitudeOnly;
```

etc (QString::compare is much more efficient vs multiple calculations of the entire lowercase string!)